### ISSUE TITLE
Cannot resubmit MFA code on Android

### ISSUE THREAD

**Issue by sameera26** [aws-amplify/amplify-android#2227]
### Before opening, please confirm:

- [ ] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin, Kotlin - Coroutines

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line
implementation 'com.amplifyframework:core:2.1.0'

```

</details>

### Environment information

<details>

```
Android


```

</details>

### Describe the bug

**Description**:

Once confirmSignIn is called with the wrong code(OTP), the library doesn't return any error response. (Using SMS MFA)

### Reproduction steps (if applicable)

```
1. Call SignIn
2. Call Confirm SignIn with the wrong code
3. SDK Doesn't return any error response

```

### Code Snippet

```kotlin
// Put your code below this line.

try {
        Amplify.Auth.confirmSignIn(
                otpNumber,
                { result ->
                    if (result.isSignedIn) {
                        LogUtil.printDebug("AuthQuickstart", "Confirm sign in succeeded: $result")
                        

                    } else {
                        LogUtil.printError(
                            "AuthQuickstart",
                            "Confirm sign in not complete. There might be additional steps: ${result.nextStep}"
                        )
                    }
                },
                { error ->
                    LogUtil.printError("AuthQuickstart", "Failed to confirm sign in - $error")
                }
            )
        } catch (error: Exception) {
            LogUtil.printError("AuthQuickstart", "Failed to confirm sign in - $error")
        }


### Log output

<details>

```
// Put your logs below this line

```

</details>


### amplifyconfiguration.json

_No response_

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by div5yesh** [comment]
@sameera26 I was unable to replicate the issue. Please attach debug logs to this issue so that we can further investigate.

**Comment by gpanshu** [comment]
Please try again with the 2.2.2 version of the library. There was another MFA related fix that was added. I believe that should fix your issue. Thank you.

**Comment by sameera26** [comment]
I tried version 2.2.2 on Android. But the issue is still exist. SDK doesn't return anything when I enter wrong OTP or auth challenge. Therefore there is no way to handle retrying `Amplify.Auth.confirmSignIn()`. Please kindly help to resolve this issue.

Also we found that iOS has reported same issue and it's already fixed [#2617](https://github.com/aws-amplify/amplify-swift/pull/2617)

Here is my code:

`
try {

            Amplify.Auth.confirmSignIn(
                "123456",
                { result ->
                    if (result.isSignedIn) {
                        LogUtil.printDebug("AuthQuickstart", "Confirm sign in succeeded: $result")
                    } else {
                        LogUtil.printError(
                            "AuthQuickstart",
                            "Confirm sign in not complete. There might be additional steps: ${result.nextStep}"
                        )
                    }
                },
                { error ->
                    LogUtil.printError("AuthQuickstart", "Failed to confirm sign in - $error")
                }
            )
        } catch (error: Exception) {
            LogUtil.printError("AuthQuickstart", "Failed to confirm sign in - $error")
        }`
